app/src/main/java/de/danoeh/antennapod/fragment/preferences/StatisticsFragment.java

```diff
@@ -1,180 +1,96 @@
-        refreshStatistics();
     }
 
-    @Override
-    public void onDestroyView() {
-        super.onDestroyView();
-        if (disposable != null) {
-            disposable.dispose();
-        }
-    }
+    public static class StatisticsPagerAdapter extends FragmentPagerAdapter {
 
-    @Override
-    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
-        inflater.inflate(R.menu.statistics, menu);
-        menu.findItem(R.id.statistics_reset).setEnabled(!countAll);
-    }
+        private final Resources resources;
+        private final Fragment[] fragments = {
+                new PlaybackStatisticsFragment(),
+                new DownloadStatisticsFragment(),
+        };
 
-    @Override
-    public boolean onOptionsItemSelected(MenuItem item) {
-        if (item.getItemId() == R.id.statistics_mode) {
-            selectStatisticsMode();
-            return true;
-        }
-        if (item.getItemId() == R.id.statistics_reset) {
-            confirmResetStatistics();
-            return true;
+        public StatisticsPagerAdapter(FragmentManager fm, Resources resources) {
+            super(fm);
+            this.resources = resources;
         }
-        return super.onOptionsItemSelected(item);
-    }
-
-    private void selectStatisticsMode() {
-        View contentView = View.inflate(getContext(), R.layout.statistics_mode_select_dialog, null);
-        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
-        builder.setView(contentView);
-        builder.setTitle(R.string.statistics_mode);
-
-        if (countAll) {
-            ((RadioButton) contentView.findViewById(R.id.statistics_mode_count_all)).setChecked(true);
-        } else {
-            ((RadioButton) contentView.findViewById(R.id.statistics_mode_normal)).setChecked(true);
-        }
-
-        builder.setPositiveButton(android.R.string.ok, (dialog, which) -> {
-            countAll = ((RadioButton) contentView.findViewById(R.id.statistics_mode_count_all)).isChecked();
-            listAdapter.setCountAll(countAll);
-            prefs.edit().putBoolean(PREF_COUNT_ALL, countAll).apply();
-            refreshStatistics();
-            getActivity().invalidateOptionsMenu();
-        });
 
-        builder.show();
-    }
-
-    private void confirmResetStatistics() {
-        if (!countAll) {
-            ConfirmationDialog conDialog = new ConfirmationDialog(
-                    getActivity(),
-                    R.string.statistics_reset_data,
-                    R.string.statistics_reset_data_msg) {
-
-                @Override
-                public void onConfirmButtonPressed(DialogInterface dialog) {
-                    dialog.dismiss();
-                    doResetStatistics();
-                }
-            };
-            conDialog.createNewDialog().show();
+        @Override
+        public Fragment getItem(int position) {
+            return fragments[position];
```

Comment:
Please create new fragments here instead of having the array. Fragments are sometimes created and deleted by the system. Maintaining a list of fragments might make the next developer access them directly from the list (which would result in invalid state).